src/sas/sascalc/data_util/registry.py

```diff
@@ -59,62 +62,56 @@ class ExtensionRegistry(object):
         registry.load('hello.cx',format='cx3') ->
             return cx3('hello.cx')
     """
-    def __init__(self, **kw):
-        self.loaders = {}
+    def __init__(self):
+        self.readers = defaultdict(list)
 
     def __setitem__(self, ext, loader):
-        if ext not in self.loaders:
-            self.loaders[ext] = []
-        self.loaders[ext].insert(0, loader)
+        self.readers[ext].insert(0, loader) # TODO: Why insert at zero, not just append?
 
     def __getitem__(self, ext):
-        return self.loaders[ext]
+        return self.readers[ext]
 
-    def __contains__(self, ext):
-        return ext in self.loaders
+    def __contains__(self, ext: str):
+        return ext in self.readers
 
     def formats(self):
         """
         Return a sorted list of the registered formats.
         """
-        names = [a for a in self.loaders.keys() if not a.startswith('.')]
+        names = [a for a in self.readers.keys() if not a.startswith('.')] # What is this doing?
         names.sort()
         return names
 
     def extensions(self):
         """
         Return a sorted list of registered extensions.
         """
-        exts = [a for a in self.loaders.keys() if a.startswith('.')]
+        exts = [a for a in self.readers.keys() if a.startswith('.')]
         exts.sort()
         return exts
 
-    def lookup(self, path):
+    def lookup(self, path: str):
         """
         Return the loader associated with the file type of path.
 
         :param path: Data file path
         :return: List of available readers for the file extension (maybe empty)
         """
+
         # Find matching lower-case extensions
         path_lower = path.lower()
-        extlist = [ext for ext in self.extensions() if path_lower.endswith(ext)]
-        # Sort matching extensions by decreasing order of length
-        extlist.sort(key=len)
+        extensions = [ext for ext in self.extensions() if path_lower.endswith(ext)]
+
+        # Sort matching extensions by decreasing order of length # TODO: Again, why???
+        extensions.sort(key=len)
+
         # Combine loaders for matching extensions into one big list
-        loaders = []
-        for L in [self.loaders[ext] for ext in extlist]:
-            loaders.extend(L)
-        # Remove duplicates if they exist
-        if len(loaders) != len(set(loaders)):
-            result = []
-            for L in loaders:
-                if L not in result:
-                    result.append(L)
-            loaders = result
-        return loaders
-
-    def load(self, path, format=None):
+        loaders = [loader for ext in extensions for loader in self.readers[ext]]
+
+        # Remove duplicates and return
+        return list(set(loaders))
+
```

Comment:
Won't this modify the ordering of the original list? It has just been sorted in line 106

Reply:
Good catch. Moving the sort to the return statement would fix this.

Reply:
yeah, true.

Reply:
fixed